resource.go

```diff
@@ -12,9 +13,16 @@ const (
 )
 
 type Config struct {
-	ClusterStatusFunc func(v interface{}) (providerv1alpha1.StatusCluster, error)
-	Logger            micrologger.Logger
-	NodeCountFunc     func(v interface{}) (int, error)
+	ClusterEndpointFunc func(v interface{}) (string, error)
+	ClusterIDFunc       func(v interface{}) (string, error)
+	ClusterStatusFunc   func(v interface{}) (providerv1alpha1.StatusCluster, error)
+	// TODO replace this with a G8sClient to fetch the node versions from the
+	// NodeConfig status once we can use the NodeConfig for general node
+	// management. As of now NodeConfig CRs are still used for draining in older
+	// guest clusters.
+	GuestCluster  guestcluster.Interface
```

Comment:
The todo maybe should be more explicit depending on what we want to have it look like now. The decision process might be still ongoing.